src/filter.rs

```diff
@@ -239,10 +229,7 @@ impl Filter {
                         + b[3] * filter_state[3]
                         + b[4] * filter_state[4];
 
-                    filter_state[4] = filter_state[3];
-                    filter_state[3] = filter_state[2];
-                    filter_state[2] = filter_state[1];
-                    filter_state[1] = filter_state[0];
+                    filter_state.copy_within(0..4, 1);
```

Comment:
Note that this is only compiled to the same code with `-C opt-level=2`. With `-C opt-level=1` it ends up not being inlined, which then keeps bounds checks etc.

Reply:
As it turns out, even if inlined it's not compiling to the same code and causes a 20-35% slowdown. See https://github.com/sdroege/ebur128/issues/52

Reverted this part for now :)

Reply:
Ok, I see. Thanks